nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -1,21 +1,146 @@
+            success, frame = self.vc.read()
+            ts2.append(self.vc.get(cv2.CAP_PROP_POS_MSEC) / 1000)
+            if not success:
+                break
+        return np.array(ts2)
+
+    def get_movie_fps(self):
+        """Return the internal frames per second (fps) for a movie file"""
+        assert self.isOpened(), self._movie_open_msg
+        prop = self.get_cv_attribute("CAP_PROP_FPS")
+        return self.vc.get(prop)
+
+    def get_frame_shape(self) -> Tuple:
+        """Return the shape of frames from a movie file."""
+        frame = self.get_movie_frame(0)
+        if frame is not None:
+            return frame.shape
+
+    @property
+    def frame_count(self):
+        if self._frame_count is None:
+            self._frame_count = self._movie_frame_count()
+        return self._frame_count
+
+    @frame_count.setter
+    def frame_count(self, val):
+        self._frame_count = val
+
+    def get_movie_frame_count(self):
+        return self.frame_count
+
+    @staticmethod
+    def get_cv_attribute(attribute_name):
+        if int(cv2.__version__.split(".")[0]) < 3:  # pragma: no cover
+            return getattr(cv2.cv, "CV_" + attribute_name)
+        return getattr(cv2, attribute_name)
+
+    def _movie_frame_count(self):
+        """Return the total number of frames for a movie file."""
+        assert self.isOpened(), self._movie_open_msg
+        prop = self.get_cv_attribute("CAP_PROP_FRAME_COUNT")
+        return int(self.vc.get(prop))
+
+    @property
+    def current_frame(self):
+        return self._current_frame
+
+    @current_frame.setter
+    def current_frame(self, frame_number):
+        assert self.isOpened(), self._movie_open_msg
+        set_arg = self.get_cv_attribute("CAP_PROP_POS_FRAMES")
+        set_value = self.vc.set(set_arg, frame_number)
+        if set_value:
+            self._current_frame = frame_number
+        else:
+            raise ValueError(f"could not set frame no {frame_number}")
+
+    def get_movie_frame(self, frame_number: int):
+        """Return the specific frame from a movie."""
+        assert self.isOpened(), self._movie_open_msg
+        assert frame_number < self.get_movie_frame_count(), "frame number is greater than length of movie"
+        self.current_frame = frame_number
+        success, frame = self.vc.read()
+        self.current_frame = 0
+        return frame
+
+    def get_movie_frame_dtype(self):
+        """Return the dtype for frame in a movie file."""
+        frame = self.get_movie_frame(0)
+        if frame is not None:
+            return frame.dtype
+
+    def release(self):
+        self.vc.release()
+
+    def isOpened(self):
+        return self.vc.isOpened()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        assert self.isOpened(), self._movie_open_msg
+        if self._current_frame < self.frame_count:
+            self.current_frame = self._current_frame
```

Comment:
There's going to be an unnecessary performance issue here by calling the setter for `current_frame` (which forcibly sets the frame position) each time we increment the count; as this is a magic method, it should be sufficient to simply increment the private reference directly and rely on the `vc.read()` to set the frame position in the capture object itself.